src/param.rs

```diff
@@ -1075,22 +1131,34 @@ impl AudioParamProcessor {
             }
         };
 
-        if is_k_rate || is_constant_block {
+        if !is_a_rate || is_constant_block {
             self.buffer.push(self.intrinsic_value);
             // nothing to compute in timeline, for both k-rate and a-rate
             if is_constant_block {
                 return;
             }
         }
 
+        // For borrow check reasons, we move move out the `event_timeline`,
+        // and pass it around as &mut.
+        let mut event_timeline = std::mem::take(&mut self.event_timeline);
```

Comment:
Just used the same strategy as in graph render to workaround the borrow checker. Maybe there is a more simple solution but I couldn't find it...

Reply:
Also, `EventTimeline` contains a `Vec`, do it allocates when replacing with default?

Reply:
Hum, I guess this is my answer: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Default-for-Vec%3CT,+Global%3E